### DISCUSSION TITLE
Compiling error in LDT

### DISCUSSION THREAD

**Discussion by MariRiba** [NASA-LIS/LISF#1506] [Q&A]
Hi everyone
I have installed LISF (7.4.2) in my system successfully and I am now trying to install NU-WRF (11.2) in the same system. 
As NU-WRF also has LISF (7.4.0)  in it , I am using the same libraries ( libraries are same for LISF v-7.4.0 and later versions ) .
I have sourced it using the same enviroment (i.e compilers ,$LD_LIBRARY_PATH and $PATH ).
Source file is as follows :

```
#!/bin/bash

module load gnu9/9.4.0

ulimit -s unlimited

export LDT_ARCH=linux_gfortran
export LDT_FC=mpif90
export LDT_CC=mpicc
export LDT_MODESMF=/scratch2/mari/lisf/test3/dep/esmf-ESMF_8_1_1/esmf_lib/mod/modO/Linux.gfortran.64.openmpi.default
export LDT_LIBESMF=/scratch2/mari/lisf/test3/dep/esmf-ESMF_8_1_1/esmf_lib/lib/libO/Linux.gfortran.64.openmpi.default
export LDT_OPENJPEG=/scratch2/mari/lisf/test3/dep/openjpeg-2.4.0/openjpeg_lib
export LDT_ECCODES=/scratch2/mari/lisf/test3/dep/eccodes-2.19.0-Source/eccodes_lib
export LDT_NETCDF=/scratch2/mari/lisf/test3/dep/netcdf-c-4.7.4/netcdf_lib
export LDT_HDF4=/scratch2/mari/lisf/test3/dep/hdf4-hdf-4_2_16/hdf4_lib
export LDT_HDF5=/scratch2/mari/lisf/test3/dep/hdf5-1.12.0/hdf5_lib
export LDT_HDFEOS=/scratch2/mari/lisf/test3/dep/hdf-eos2-3.0/hdfeos_lib
export LDT_GDAL=/scratch2/mari/lisf/test3/dep/gdal-2.4.4/gdal_lib
export LDT_FORTRANGIS=/scratch2/mari/lisf/test3/dep/fortrangis-2.7-1/fortrangis_lib
export LDT_LIBGEOTIFF=/scratch2/mari/lisf/test3/dep/libgeotiff-1.6.0/libgeotiff/libgeotiff_lib
export LDT_JPEG=/scratch2/mari/lisf/test3/dep/jpeg-8d/jpeg_lib

export LIS_ARCH=linux_gfortran
export LIS_FC=mpif90
export LIS_CC=mpicc
export LIS_MODESMF=/scratch2/mari/lisf/test3/dep/esmf-ESMF_8_1_1/esmf_lib/mod/modO/Linux.gfortran.64.openmpi.default
export LIS_LIBESMF=/scratch2/mari/lisf/test3/dep/esmf-ESMF_8_1_1/esmf_lib/lib/libO/Linux.gfortran.64.openmpi.default
export LIS_OPENJPEG=/scratch2/mari/lisf/test3/dep/openjpeg-2.4.0/openjpeg_lib
export LIS_ECCODES=/scratch2/mari/lisf/test3/dep/eccodes-2.19.0-Source/eccodes_lib
export LIS_NETCDF=/scratch2/mari/lisf/test3/dep/netcdf-c-4.7.4/netcdf_lib
export LIS_HDF5=/scratch2/mari/lisf/test3/dep/hdf5-1.12.0/hdf5_lib
export LIS_HDF4=/scratch2/mari/lisf/test3/dep/hdf4-hdf-4_2_16/hdf4_lib
export LIS_HDFEOS=/scratch2/mari/lisf/test3/dep/hdf-eos2-3.0/hdfeos_lib

export LD_LIBRARY_PATH=/scratch2/mari/lisf/test3/dep/esmf-ESMF_8_1_1/esmf_lib/lib/libO/Linux.gfortran.64.openmpi.default:/scratch2/mari/lisf/test3/dep/eccodes-2.19.0-Source/eccodes_lib/lib64:/scratch2/mari/lisf/test3/dep/netcdf-c-4.7.4/netcdf_lib/lib:/scratch2/mari/lisf/test3/dep/jpeg-8d/jpeg_lib/lib:/scratch2/mari/lisf/test3/dep/openjpeg-2.4.0/openjpeg_lib/lib:/scratch2/mari/lisf/test3/dep/hdf5-1.12.0/hdf5_lib/lib:/scratch2/mari/lisf/test3/dep/fortrangis-2.7-1/fortrangis_lib/lib:/scratch2/mari/lisf/test3/dep/gdal-2.4.4/gdal_lib/lib:$LD_LIBRARY_PATH

export OMPI_MCA_btl_openib_allow_ib=true
```
But during compilation of  ldt , I am getting this error :
```
[INFO] Compiling LDT source code

Generating dependencies for ../core/LDT_timeMgrMod.F90
mpif90 -c -pass-exit-codes -ffree-line-length-0 -O2   -fconvert=big-endian -DGFORTRAN -I/scratch2/mari/lisf/test3/dep/esmf-ESMF_8_1_1/esmf_lib/mod/modO/Linux.gfortran.64.openmpi.default  -I/scratch2/mari/lisf/test3/dep/eccodes-2.19.0-Source/eccodes_lib/include/ -I/scratch2/mari/lisf/test3/dep/netcdf-c-4.7.4/netcdf_lib/include/  -I/scratch2/mari/lisf/test3/dep/hdf-eos2-3.0/hdfeos_lib/include/  -I/scratch2/mari/lisf/test3/dep/hdf4-hdf-4_2_16/hdf4_lib/include/  -I/scratch2/mari/lisf/test3/dep/hdf5-1.12.0/hdf5_lib/include/  -I/scratch2/mari/lisf/test3/dep/fortrangis-2.7-1/fortrangis_lib/include/ -I/scratch2/mari/lisf/test3/dep/fortrangis-2.7-1/fortrangis_lib/include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_timeMgrMod.F90
f951: Fatal Error: Reading module ‘ldt_priv_rcmod’ at line 1 column 2: Unexpected EOF
compilation terminated.
make: *** [Makefile:108: LDT_timeMgrMod.o] Error 1

[ERR] Compile failed
-------------------------------------------------------------------
```
Please help me how to solve this error or any suggestion on how to go about installing NU-WRF .

Thanks

**Comment by jvgeiger** [comment]
Hello.  Have you asked the NU-WRF team about this problem?  I have not compiled NU-WRF in a long time.  It will take me some time to try to replicate your error.

**Reply by MariRiba** [reply]
Hi
I asked them and they said to install the dependent libraries in a particular order as given in the documentation of NU-WRF , which is somehow a bit different from that of LISF. I tried installing in those order but i am having problems in installing the libraries again. So I tried using the same libraries as NU-WRF  also has LISF in it. 

Also one more question retated to soil moisture LISF simulation.
What is the best value to use as initial soil moisture value before starting my spinup ? 
In the target test cases files , it is given 0.4 .

Thanks

**Comment by jvgeiger** [comment]
Hello,

I do not have the latest NU-WRF source code.  The documentation that I see online only lists the required libraries in alphabetical order.  I do not see any other instructions for installing the libraries.  In which order does the NU-WRF documentation suggest?

Your error suggests that the ldt/core/LDT_PRIV_rcMod.F90 file did not compile correctly.  This file depends only on the ESMF library.  Would you please attach a log of all the messages you get when trying to build NU-WRF?

Our @NASA-LIS/lsm  team can better answer your question regarding soil moisture initialization.


Jim

**Reply by MariRiba** [reply]
Hi,
I have attached two files "baselibs.cfg" and "baselib.sh" . In baselibs.sh , they have mentioned that the installation must occur in the order provided in this script to resolve inter-library dependencies. And the list is given in file baselibs.cfg.
[baselibs.sh.txt](https://github.com/NASA-LIS/LISF/files/14887124/baselibs.sh.txt)
[baselibs.cfg.txt](https://github.com/NASA-LIS/LISF/files/14887127/baselibs.cfg.txt)

**Reply by jvgeiger** [reply]
Hello.  If I understand correctly, you tried to install the baselibs, but you had problems.  Then you tried using libraries that you previously installed for LISF.  You must be careful.  From what I see in your baselibs.sh and baselibs.cfg files and your environment settings, some of the libraries have different versions.  I suggest that you install libraries for NU-WRF using the baselibs scripts.

Please try starting over and using the baselibs scripts to install libraries.  Then set your environment variables to point to the libraries installed by the baselibs scripts.

**Reply by MariRiba** [reply]
Hi 
yes, you are correct . But I am not sure how to install using the baselibs scripts to install libraries. I can try installing normally in the provided order with the given versions.
